lib/gringotts/gateways/mercadopago.ex

```diff
@@ -0,0 +1,272 @@
+
+  ### Authorization for old customer.
+    The following example shows how one would (pre) authorize a payment of 42 BRL on a sample `card`.
+    Mention `customer_id`.  
+      iex> amount = Money.new(42, :BRL)
+      iex> card = %Gringotts.CreditCard{first_name: "Hermione", last_name: "Granger", number: "4509953566233704", year: 2099, month: 12, verification_code: "123", brand: "VISA"}
+      iex> opts = [email: "hermione@granger.com", order_id: 123125, customer_id: "308537342-HStv9cJCgK0dWU", payment_method_id: "visa", config: %{access_token: YOUR_ACCESS_TOKEN, public_key: YOUR_PUBLIC_KEY}]
+      iex> {:ok, auth_result} = Gringotts.authorize(Gringotts.Gateways.Mercadopago, amount, card, opts)
+      iex> auth_result.id # This is the authorization ID
+      iex> auth_result.token # This is the customer ID/token
+
+  """
+
+  @spec authorize(Money.t(), CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def authorize(amount , %CreditCard{} = card, opts) do
+    if Keyword.has_key?(opts, :customer_id) do
+      auth_token(amount, card, opts, opts[:customer_id], false)
+    else
+      case create_customer(opts) do
+        {:error, res} -> {:error, res}
+        {:ok, customer_id} ->  auth_token(amount, card, opts, customer_id, false)
+      end
+    end
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+  
+  # Makes the request to mercadopago's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+  @spec commit(atom, String.t(), String.t(), keyword, keyword) :: {:ok | :error, Response.t()}
+  defp commit(method, path, body, opts, url_params) do
+    headers = [{"content-type", "application/json"}, {"accept", "application/json"}]
+    HTTPoison.request(method, "#{@base_url}#{path}", body, headers, url_params) |> respond(opts)
+  end
+
+  # Parses mercadopago's response and returns a `Gringotts.Response` struct
+  # in a `:ok`, `:error` tuple.
+
+  defp auth_token(amount, %CreditCard{} = card, opts, customer_id, capture) do
+    opts = opts ++ [customer_id: customer_id]
+    {state, res} = create_token(card, opts)
+    if state == :error do
+      {state, res}
+    else
+      auth(amount, card, opts, res, capture)
+    end
+  end
+
+  defp auth(amount, %CreditCard{} = card, opts, token_id, capture) do
+    {_, value, _, _} = Money.to_integer_exp(amount)
+    opts = opts ++ [token_id: token_id]
+    url_params = [access_token: opts[:config][:access_token]]
+    body = authorize_params(value, card, opts, opts[:token_id], opts[:customer_id], capture) |> Poison.encode!
+    commit(:post, "/v1/payments", body, opts, params: url_params )
+  end
+  
+  defp create_customer(opts) do
```

Comment:
You can move that `if` here to check if the caller supplied a `customer_id` in `opts`